packages/react/src/composite/SegmentedControl/src/components/SegmentedControlButton/SegmentedControlButton.tsx

```
@@ -55,25 +69,36 @@ const SegmentedControlButton = forwardRef(
+      classnames.appearance[isSelected ? "selected" : "default"],
     ]
-      .concat(fullWidth && classnames.fullWidth ? classnames.fullWidth : "")
-      .filter(Boolean)
+      .concat(shouldUseFullWidth ? classnames.fullWidth : "")
       .join(" ");
 
     return (
       <As
-        id={`segment-${ariaID}`}
+        id={id}
         role="button"
```

Biome (2.1.2)

[error] The elements with this role can be changed to the following elements:
<button>

For examples and more information, see WAI-ARIA Roles

(lint/a11y/useSemanticElements)